### ISSUE TITLE
Consumers can't debug library built using ng-packagr, because of Missing sourcemap files

### ISSUE THREAD

**Issue by bharathmuppa** [ng-packagr/ng-packagr#2958]
## Type of Issue

```
[x] Bug Report
[ ] Feature Request
```

## Description

After publishing the library, the consumers of the library are unable to debug due to the source maps pointing to the local paths in the `fesm2022` folder. Even with settings like `inlineSources: true` in `tsconfig.lib.json` and other configurations, the generated source maps are referencing paths like `../../projects/enterprise-components/...` which are not available in the consuming project's environment.

The issue appears to be that the source map generation is not aligning with the published environment. This makes it impossible for consuming developers to trace issues back to the original source, significantly reducing the library's usability in development and debugging scenarios.

---

#### How To Reproduce

1. Use the following `ng-package.json` configuration:
   ```json
   {
     "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
     "dest": "../../../dist/common-components/material",
     "assets": [
       {
         "glob": "**/*.scss",
         "input": "src/themes",
         "output": "themes"
       }
     ],
     "lib": {
       "entryFile": "src/public-api.ts",
       "styleIncludePaths": ["src/themes"]
     },
     "allowedNonPeerDependencies": ["angular", "zone", "moment"]
   }
   ```

2. Use the following `tsconfig.lib.json`:
   ```json
   {
     "extends": "../../../tsconfig.json",
     "compilerOptions": {
       "outDir": "../../../out-tsc/lib",
       "declaration": true,
       "esModuleInterop": true,
       "inlineSources": true,
       "types": ["node"],
       "lib": ["dom", "es2018"]
     },
     "angularCompilerOptions": {
       "annotateForClosureCompiler": true,
       "skipTemplateCodegen": true,
       "strictMetadataEmit": true,
       "fullTemplateTypeCheck": true,
       "strictInjectionParameters": true,
       "enableResourceInlining": true,
       "compilationMode": "partial"
     },
     "exclude": ["src/test.ts", "**/*.spec.ts"]
   }
   ```

3. Build the library using `ng-packagr`:
   ```bash
   npm run build
   ```

4. Publish the library and use it in a different Angular project.

5. Attempt to debug the library in the consuming project.

---

#### Expected Behaviour

- The generated source maps should allow debugging in the consuming project.
- The `sources` paths in the `.map` files should point to relative paths or include `sourcesContent` inline, making debugging possible even without the local source files.

---

#### Actual Behaviour

- The `sources` paths in the `.map` files point to local development paths, such as:
  ```
  ../../projects/common-components/material/src/...
  ```
- These paths are not accessible from the consuming project, rendering the source maps unusable for debugging.

---

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr: 18.x
@angular/*: 18.x
typescript: 5..54
node: 19.x
```

---

Let me know if you need further assistance clarifying or revising this issue!

**Comment by alan-agius4** [comment]
In your application project, are you enabling vendor source maps? Without this option, the source maps of `node_modules` will not be processed. 

For more details, refer to see: https://angular.dev/reference/configs/workspace-config#source-map-configuration

**Comment by bharathmuppa** [comment]
Yes, they do enabled this. Got the confirmation and i am using it in test project as well, core problem is sourcemap links to original folder structure, rather pointing to esm or content itself.

**Comment by alan-agius4** [comment]
@bharathmuppa, I tried this locally and the sourcemaps worked correctly when `vendor` option is enabled. Can you setup a minimal repro please?

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by alan-agius4** [comment]
Closing this issue as we are unable to reproduce it. If the problem persists, please provide a minimal reproduction.
